Changes before creation of resource are not permitted, demo app crashes.

**Describe the bug**

```

Work [ id=0f8a41a1-cc23-4ccf-8012-7d5a25fdb0d0, tags={ com.google.android.fhir.demo.data.DemoFhirSyncWorker } ] failed because it threw an exception/error (Ask Gemini)
                                                                                                    java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Changes before creation of resource are not permitted
                                                                                                    	at androidx.work.impl.utils.futures.AbstractFuture.getDoneValue(AbstractFuture.java:516)
                                                                                                    	at androidx.work.impl.utils.futures.AbstractFuture.get(AbstractFuture.java:475)
                                                                                                    	at androidx.work.impl.WorkerWrapper$2.run(WorkerWrapper.java:317)
                                                                                                    	at androidx.work.impl.utils.SerialExecutorImpl$Task.run(SerialExecutorImpl.java:96)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
                                                                                                    	at java.lang.Thread.run(Thread.java:1012)
                                                                                                    Caused by: java.lang.IllegalArgumentException: Changes before creation of resource are not permitted
                                                                                                    	at com.google.android.fhir.sync.upload.patch.PerResourcePatchGenerator.mergeLocalChangesForSingleResource(PerResourcePatchGenerator.kt:66)
                                                                                                    	at com.google.android.fhir.sync.upload.patch.PerResourcePatchGenerator.generateSquashedChangesMapping$engine_debug(PerResourcePatchGenerator.kt:50)
                                                                                                    	at com.google.android.fhir.sync.upload.patch.PerResourcePatchGenerator.generate(PerResourcePatchGenerator.kt:41)
                                                                                                    	at com.google.android.fhir.sync.upload.Uploader.upload(Uploader.kt:58)
                                                                                                    	at com.google.android.fhir.sync.FhirSynchronizer$upload$2.invoke(FhirSynchronizer.kt:136)
                                                                                                    	at com.google.android.fhir.sync.FhirSynchronizer$upload$2.invoke(FhirSynchronizer.kt:136)
                                                                                                    	at com.google.android.fhir.impl.FhirEngineImpl$syncUpload$2.invokeSuspend(FhirEngineImpl.kt:154)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:811)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:715)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:702)

```
**Component**
Demo app.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to CRUD operation screen. 
2. Create Patient by clicking on create button
3. Repeat the step 2 by clicking create button some more time.
4. Go back to the dashboard screen
5. Click on one time sync, perform sync, app get crashed.
**Expected behavior**
App should not be crashed.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Smartphone (please complete the following information):**
 - Device: Nokia 3.4
 - Android version: 12
 - Build number: [e.g. Settings -> About phone -> Build number]
 - Version [e.g. 22]

**Additional context**
Add any other context about the problem here.

**Would you like to work on the issue?**
Please state if this issue should be assigned to you or who you think could help to solve this issue.

--- a/demo/src/main/java/com/google/android/fhir/demo/CrudOperationFragment.kt
+++ b/demo/src/main/java/com/google/android/fhir/demo/CrudOperationFragment.kt
@@ -304,6 +304,7 @@ class CrudOperationFragment : Fragment() {
         gender = it.gender,
         isActive = it.isActive,
       )
+      setupUiForCrudOperation(OperationType.CREATE)
     }
   }
 

